File: day-01-binary-search/python/binary-search.py
def binary_search(lst, item):
    low = 0
    high = len(lst) - 1

    test = 0
    while low <= high:
        mid = (low + high) // 2
        guess = lst[mid]
        test += 1

        if guess == item:
            return mid, test
        if guess > item:
            high = mid - 1
        else: 
            low = mid + 1

    return None, test

my_list = list(range(1, 1025))  

index, tries = binary_search(my_list, 512)
if index is not None:
    print(f"Found at index {index} in {tries} tries")
else:
    print(f"Not found after {tries} tries")

File: day-01-binary-search/js/binary-search.js
function binarySearch(arr, item) {
  let low = 0;
  let high = arr.length - 1;
  let steps = 0;

  while (low <= high) {
    steps++;
    let mid = Math.floor((low + high) / 2),
      guess = arr[mid];

    if (guess === item) {
      return { index: mid, steps };
    } else if (guess > item) {
      high = mid - 1;
    } else {
      low = mid + 1;
    }
  }
  return { index: null, steps };
}

const myArr = Array.from({ length: 1024 }, (_, i) => i + 1);

console.log(binarySearch(myArr, 512));
